경북대 BE 이원준 1단계 - 엔티티 매핑

lwj0831 opened the pull request:
## ✅ 해결하려는 문제

1. 이번 요구사항은 WishItem, Member, Product 등의 도메인에서 JPA 기반 Repository로 리팩토링하는 것이었습니다.  
기존의 도메인 클래스를 엔티티 클래스로 전환하게 되면 JDBC 기반 Repository에서는 더 이상 재사용할 수 없는 문제가 발생했으며,  
단순히 JDBC Repository를 제거하는 것이 아닌, JPA에 의존하지 않고 다양한 DB 접근 기술로 교체할 수 있는 유연한 구조를 만들고자 했습니다.

2. 환경별로 유연한 설정을 적용할 수 있도록 Spring Boot의 프로파일 기반 설정을 학습하고 적용하고자 했습니다.  
특히 `init.mode`, `ddl-auto`, `schema.sql`, `data.sql`, `spring.jpa.defer-datasource-initialization` 등 설정 간의 관계를 명확히 이해하지 못해  예상치 못한 초기화 문제를 겪었고, 이를 해결하며 내부 동작 흐름을 익혔습니다.

3. `JpaRepository`를 상속하는 `JpaRepositoryImpl` 클래스를 정의했을 때,  
Spring Boot가 컴포넌트 어노테이션 없이도 해당 클래스를 빈으로 등록하려고 시도하여 중복 빈 등록 오류가 발생했습니다.  
이를 통해 Spring Data JPA의 자동 구현 클래스 탐지 및 빈 등록 방식에 대해 심도 있게 학습할 수 있었습니다.

4. 페이징 정렬 필드와 같이 공통/도메인별로 분리되는 유효성 검증 로직을 어디서 어떻게 수행할 것인지에 대한 기준을 명확히 세우는 데 고민이 있었습니다.  
특히 페이징 정렬 필드는 비즈니스 도메인에 종속적인 성격도 있으면서, 동시에 UI/페이징과도 관련되어 있어 구조적 위치 선정이 고민되었습니다.

---

## ✅ 어떻게 해결했는지

1. JDBC 기반 Repository를 유지하기 위해 도메인 객체와 엔티티 객체를 분리하고, Mapper 클래스를 통해 변환하도록 설계했습니다.  
이로 인해 각 Repository 구현체는 자신이 사용하는 기술 스택에 맞게 독립적인 구조를 가질 수 있게 되었습니다.

2. `JpaRepository`를 직접 상속하는 구현체에 대해 Spring이 자동으로 빈 등록을 시도하는 문제를 피하기 위해,  
클래스명을 `JpaWishItemRepositoryAdapter`처럼 `Adapter`로 명명하여 `JpaRepository`와 구분되도록 했습니다.  
해당 방식이 명시적으로 역할을 드러내고 빈 충돌을 방지하는 데에는 효과적이었지만,  
이러한 방식이 최선인지 또는 더 좋은 설계 패턴이 있는지 궁금합니다.

3. 프로파일에 따라 `dev`(JPA+MySQL), `test`(JPA+H2), `default`(JDBC+H2) 환경이 자동 전환되도록 구성했습니다.  
각 설정에 따라 Repository 및 DB 초기화 스크립트가 올바르게 작동하도록 `@Profile`과 `application.yml`을 설정했습니다.

4. `ascending` 여부는 공통적인 로직이므로 ArgumentResolver에서 처리하고,  
`sortField`는 도메인마다 유효 필드가 다르므로 Service 계층에서 검증하도록 변경했습니다.

---

## ✅ 이 부분을 중점적으로 리뷰해주세요

1. `update`, `delete` 시 엔티티 존재 여부 검증을 현재는 Service 계층에서 처리하고 있는데,  
이 검증 책임을 Repository 쪽으로 위임하는 것이 더 나은지 고민이 됩니다.  
역할 분리에 대한 피드백을 주시면 감사하겠습니다.

2. 페이징 정렬 필드인 `sortField` 유효성 검증을 Service에서 처리하고 있으나,  
비즈니스 로직과 UI 페이징 로직이 혼재되어 구조적으로 어색한 면이 있습니다.  
ArgumentResolver와 같은 Controller단에서 해결, Validator를 사용한 Service단에서 해결 등 어느 계층이 가장 적절한 위치일지 의견 부탁드립니다.

3. 도메인과 엔티티를 분리한 구조는 DB 기술 추상화와 유연성 확보라는 장점이 있는 반면,  
- 매핑 로직 증가  
- 영속성 컨텍스트 관련 기능(dirty checking, cascade 등) 제약  
- 도메인/엔티티 간 불일치  
등의 복잡성을 초래합니다. 이러한 구조에서 생기는 트레이드오프를 실무적으로 어떻게 해결하는지에 대한 의견도 듣고 싶습니다.

lwj0831 committed:
build: Spring data jpa 의존성 추가

lwj0831 committed:
chore: application.yml 분리 및 업데이트

- dev, test 프로파일에 따른 application.yml분리
- application-dev.yml MYSQL 데이터 소스 정보 추가
- application-test.yml H2 inmemory DB 데이터 소스 사용

lwj0831 committed:
build: MYSQL driver 의좃넝 추가

lwj0831 committed:
feat(product): ProductEntity 정의 및 ProductMapper 구현

- 엔티티 클래스 ProductEntity 정의
- Product 도메인 모델과 ProductEntity 사이의 매핑 헬퍼 클래스
  ProductMapper 정의

lwj0831 committed:
feat(product): JpaProdcutRepository 추가 및 구현

- JPA기반 ProductEntity 레포지토리 역할을 담당하는 JpaProductRepository
  정의
- JpaProductRepository를 사용하는 ProductRepository인터페이스 구현체 jpaProductRepositoryAdapter 구현

lwj0831 committed:
chore(config): ProductRepositoryConfig 업데이트

- dev,test 프로파일에서 JpaProductRepository구현체 사용하도록 변경

lwj0831 committed:
chore: Datasource url 오류 수정

-대신 underscore(_)으로 데이터베이스명 변경

lwj0831 committed:
chore: dev프로파일용 data-dev.sql 추가

lwj0831 committed:
chore(product): ProductEntity 엔티티 테이블명, id컬럼명 변경

- 기존의 h2, jdbc기반 레포지토리 컬럼명과의 일관성을 위해 테이블명 products(복수형)->product(단수형)으로 변경
- 편의성을 위해 id컴럼명 product_id -> id로 변경

lwj0831 committed:
refactor: Pagination 정렬 필드 값 검증 리팩토링

- Product 도메인 기준 정렬 필드값 검증 서비스 계층에서 수행하도록
  리팩토링
- 유효한 상품 도메인 정렬 필드값을 위한 ValidProductSortFields 구현
- 불필요한 객체 생성 방지를 위해 기존 InvalidProductWords 미리 생성한 객체 사용하도록 리팩토링
- 상품관련 검증 담당하는 ProductValidator내 validateProductSortField
  로직 추가
- ProductService내 정렬 필드 검증 로직 사용하도록 추가
- ProductSortStrategyFactory내 정렬 필드 검증 로직 삭제하여 중복 검증
  방지

lwj0831 committed:
chore: ProductExceptionHandler, ProductApiStructuredE2ETest테스트 코드 수정

lwj0831 committed:
chore: 프로파일별 yml 설정 및 datasource 초기화 스크립트 수정

- dev, test 프로파일 환경에서 Jpa기반 ProductRepository사용
- default 프로파일 환경에서 jdbc기반 ProductRepository사용
- Spring boot가 별도의 data-loactions, schema-loactions지정 안할 시
  기본값으로 data.sql, schema.sql 사용하는 문제 환경별로 명확하게 리네이밍하여 개선
- default 프로파일에서 사용되는 data-default.sql, schema-defult.sql 추가

lwj0831 committed:
feat(member): MemberEntity 정의 및 MemberMapper헬퍼 클래스 정의

- 회원 엔티티 클래스 MemberEntity 정의
- Member - MemberEntity 매핑 헬퍼 클래스 MemberMapper 정의

lwj0831 committed:
feat(member): JPA기반 MemberRepository 구현

lwj0831 committed:
feat(memberAuth): MemberAuthEntity 정의, MemberAuthMapper 엔티티-도메인 간 변환 헬퍼 클래스 정의

lwj0831 committed:
feat(memberAuth): Jpa기반 MemberAuthRepository 구현

lwj0831 committed:
refactor(wishItem): wishItem 객체 참조 방식으로 변경

- 객체지향방식 적용을 위해 기존의 memberId, productId -> Member, Product 참조하도록 변경
- JdbcWishItemRepository, WishItemService 변경사항 반영

lwj0831 committed:
chore: SimpleWishItemDto Product->WishItem도메인으로 패키지 위치 변경

lwj0831 committed:
feat(wishItem): WishItem엔티티 클래스 및 WishITemMapper 헬퍼 클래스 도입

lwj0831 committed:
feat(wishItem): JPA기반 WishItemRepository 구현

- 커스텀 쿼리와 fetch join 사용하여 N+1문제 방지
- WishItemRepository 인터페이스에 맞게 구현

lwj0831 committed:
style: 코드 리포맷팅

lwj0831 committed:
chore(wishItem): Primary를 통해 JpaWishItemRepository 빈등록 우선 순위 설정

lwj0831 committed:
docs: STEP3 1단계 구현 사항 작성

lwj0831 committed:
test: JpaMemberAuthRepositoryAdapter, JpaProductRepositoryAdapter 단위 테스트 추가

lwj0831 committed:
test: JpaWishItemRepositoryAdapter단위 테스트 추가

lwj0831 committed:
style: Code reformatting

Rok93 commented on src/main/java/gift/auth/jpa/MemberAuthEntity.java:
```diff
@@ -0,0 +1,56 @@
+
+@Entity
+@Table(name = "member_auth")
+public class MemberAuthEntity {
```
원준님은 도메인 객체와 JPA의 Entity를 분리하시는 선택을 하셨는데, 
전 개인적으로 말씀하신 장/단점을 따져봤을 때, 지금의 구조에서는 굳이 두 개념을 쪼개지 않았을 것 같은데, 쪼개신 이유가 있는지가 궁금합니다.  🤔

Rok93 replied:
남겨주신 질문과 관련이있어보여서 여기에서 추가로 제 생각을 남겨볼게요.  😃 

> 도메인과 엔티티를 분리한 구조는 DB 기술 추상화와 유연성 확보라는 장점이 있는 반면... 도메인/엔티티 간 불일치 등의 복잡성을 초래합니다. 이러한 구조에서 생기는 트레이드오프를 실무적으로 어떻게 해결하는지에 대한 의견도 듣고 싶습니다.

먼저 실무에서도 생각보다 순수 도메인 객체와 JPA Entity를 분리하는 방식을 취하는 조직이 적은 것 같습니다. 
말씀하신 것처럼 이렇게 분리를하게되면 번거로운 절차들이 많이 생기기 때문에, 이런 불편함을 고려하여 트레이드 오프를 따져봤을 때, 대다수의 경우에는 `도메인 객체 = 엔티티`로 생각하는 의사결정을 하는 것 같습니다. 

해결은 할 수 없습니다. 분리하면 어쨋거나 Domain <-> Entity 변환이 거의 모든 경우에 필요해지기 때문에 이런 변환 계층이 추가되게되고 계층이 추가된다는 것은 시스템이 복잡해진다는 의미가됩니다.

> 트레이드오프를 실무적으로 어떻게 해결하는지

조금 엄밀하게말하면 트레이드오프라는 표현부터 바로 잡을 필요가 있다고 생각해요. 
A, B 안이 있을 대, A안도 장/단점이있고 B안도 장/단점이 있을텐데요. 
결국 A안도 B안도 절대적인 답은 없습니다. 
하지만 상황이 놓인다면 A안, B안 중 조금 더 좋은 안이 있다는 사실은 명확해지는 것 같아요. 
장황하게 설명했지만, 결국 이럼에도 말씀하신 장점들이 더 중요한 상황이라면 도입할 수 있다고 생각합니다. 

사실 저도 분리하는 방식을 실무에서 거의 써본적이 없어서 아마 원하시는 답변을 명확하게 해드리지는 못할 것 같습니다.

Rok93 replied:
`아래는 해결하려는 문제`라고 말씀하셨는데요. 
이렇게 규모에 비해서 불필요한 분리 때문에 오히려 생산성이 저하될 것 같다는 우려가 생기는데요. 
저희가 JPA를 다른 ORM 혹은 다른 영속성 라이브러리(?)로 대체할 일이 자주 발생할까요? 
이게 문제라고 생각할 수 있을까요?

> 이번 요구사항은 WishItem, Member, Product 등의 도메인에서 JPA 기반 Repository로 리팩토링하는 것이었습니다.
기존의 도메인 클래스를 엔티티 클래스로 전환하게 되면 JDBC 기반 Repository에서는 더 이상 재사용할 수 없는 문제가 발생했으며,
단순히 JDBC Repository를 제거하는 것이 아닌, JPA에 의존하지 않고 다양한 DB 접근 기술로 교체할 수 있는 유연한 구조를 만들고자 했습니다.

lwj0831 replied:
> 원준님은 도메인 객체와 JPA의 Entity를 분리하시는 선택을 하셨는데, 전 개인적으로 말씀하신 장/단점을 따져봤을 때, 지금의 구조에서는 굳이 두 개념을 쪼개지 않았을 것 같은데, 쪼개신 이유가 있는지가 궁금합니다. 🤔

사실 기존의 Jdbc 레포지토리를 유지하면서 Jpa기반 레포지토리도 도입해보고 싶다는 생각에 도메인과 엔티티 클래스를 분리하였습니다. 예전부터 도메인과 엔티티를 분리하는 구조를 사용하는 경우가 많을까 궁금하기도 했고 관련해서 피드백 받아보고 싶어서 이 방식으로 구현해봤습니다! 

> `아래는 해결하려는 문제`라고 말씀하셨는데요. 이렇게 규모에 비해서 불필요한 분리 때문에 오히려 생산성이 저하될 것 같다는 우려가 생기는데요. 저희가 JPA를 다른 ORM 혹은 다른 영속성 라이브러리(?)로 대체할 일이 자주 발생할까요? 이게 문제라고 생각할 수 있을까요?

권장 PR 요청문에 맞게 작성하다보니 오해의 소지가 살짝 생긴 것 같습니다..! '해결하려는 문제'보다 '구현 배경' 정도로 봐주시면 좋을 것 같습니다!
도메인, 엔티티 분리 구조로 얻을 수 있는 장점이라고 하면 DB구현 기술을 교체하기에는 용이하다고 생각했습니다. 그러나 JPA이외의 다른 ORM기술로의 교체하는 일은 자주 일어나지 않을 것이라는 의견에 저도 동의합니다!

제가 생각하기에도 지금 구현사항으로는 오히려 복잡성만 증가하는 좋지 않은 구조라고 판단되어서 Jdbc관련 레포지토리는 깔끔하게 삭제하고 Jpa기반 레포지토리를 사용하도록 리팩토링하였습니다 😊한번쯤 피드백 받아보고 싶은 내용에 대한 현실적인 답변을 들을 수 있어서 좋았습니다!

Rok93 commented on src/main/java/gift/auth/jpa/MemberAuthEntity.java:
```diff
@@ -0,0 +1,56 @@
+  private String password;
+  private String refreshToken;
+
+  public MemberAuthEntity() {
+  }
```
JPA 사용을 위해서는 기본 생성자가 필요한데요. 😃
왜 필요한지도 이번 기회에 알아가면 좋을 것 같아요. 

추가적으로 의도한 방식이 아니기 때문에 MemberAuthEntity 객체 생성할 때, 기본 생성자를 못쓰도록 최대한 막는 것이 좋다고 생각해요. 
하지만 기술적으로 기본생성자를 공개할 수 밖에 없는 상황이니 접근 제한자를 `protected` 로 제한해보면 어떨까요? 😉

lwj0831 replied:
이 부분은 캐치하지 못한 것 같습니다! Jpa에서는 리플렉션을 통해 엔티티 객체를 생성하는 것으로 알고 있는데 이를 위해 기본 생성자가 필수적으로 필요하다고 알고 있습니다. private는 안되는 건가 공부해보았는데요! 직렬화 및 호환성 문제가 있다고 해서 말씀해주신대로 protected를 사용하도록 변경했습니다😊

Rok93 commented on src/main/java/gift/auth/jpa/MemberAuthEntity.java:
```diff
@@ -0,0 +1,56 @@
+    this.id = memberAuth.memberId();
+    this.email = memberAuth.email().getEmailText();
+    this.password = memberAuth.password();
+    ;
```
불필요한 `;` 인 것 같네요. 제거 부탁드려요. 👀🦅

Rok93 commented on src/main/java/gift/auth/jpa/MemberAuthMapper.java:
```diff
@@ -0,0 +1,17 @@
+
+import gift.auth.domain.MemberAuth;
+
+public class MemberAuthMapper {
```
이런 Mapper는 객체를 생성할 필요가 없으니 기본 생성자를 private으로 제한하여 객체 생성을 방지해주면 좋을 것 같아요.

lwj0831 replied:
유틸리티성 클래스라 객체 생성 방지가 필요한 것 같습니다!  상속도 불가능하게 final 클래스로 선언하는 것도 좋은 것 같다고 생각합니다
리팩토링하면서 해당 클래스가 필요없어져서 삭제하는 방향으로 리팩토링했습니다

Rok93 commented on src/main/java/gift/auth/repository/JpaMemberAuthRepositoryAdapter.java:
```diff
@@ -0,0 +1,55 @@
+  private final JpaMemberAuthRepository jpaMemberAuthRepository;
+
+  public JpaMemberAuthRepositoryAdapter(JpaMemberAuthRepository jpaMemberAuthRepository) {
+    this.jpaMemberAuthRepository = jpaMemberAuthRepository;
```
[프로그래밍 요구사항](https://edu.nextstep.camp/s/UH6H8Iba/ls/Mvfn9N7o)을 지키며 미션을 수행하는 것이 전반적인 미션 요구사항인데요. 아래와 같은 요구사항이 있는데, 코드 전반적으로 반영이 안된 것 같네요. 

IDE 셋팅으로 코드 스타일을 수정해주면 이 변경을 직접 하실 필요 없이 일괄적으로 진행하실 수 있으니 관련 설정을 알아보시고 적용해보시면 좋을 것 같습니다. 

> 단, 들여쓰기는 '2 spaces'가 아닌 '4 spaces'로 한다.

lwj0831 replied:
제가 꼼꼼하게 확인하지 못한 것 같습니다...! IDE 코드 스타일 수정했습니다 😂

Rok93 commented on src/main/java/gift/auth/repository/JpaMemberAuthRepositoryAdapter.java:
```diff
@@ -0,0 +1,55 @@
+  }
+
+  @Override
+  public void update(Long memberId, MemberAuth updatedMemberAuth) {
```
> update, delete 시 엔티티 존재 여부 검증을 현재는 Service 계층에서 처리하고 있는데,
이 검증 책임을 Repository 쪽으로 위임하는 것이 더 나은지 고민이 됩니다.
역할 분리에 대한 피드백을 주시면 감사하겠습니다.

update나 delete하는 쿼리를 수행했을 때(주어진 조건절에 대해서) 일단 수행이 됐다면 저는 Repository는 제 역할을 다했다고 생각해요.  그렇기 때문에 Repository에서 검증할 부분이 아닌 Service단에서 처리하는 것이 좋다고 생각해요. 

예를들어 없는 자원에 대한 삭제 요청에 대해서 우리 서비스에서는 예외응답 없이 그냥 정상 처리하겠다! 라는 정책이 있을 수도 있을텐데요. 이런 비즈니스 정책이 바뀜에따라 대응해야하는 곳은 Repository여야할까요? 아니면 Service 쪽이어야할까요?

lwj0831 replied:
'비즈니스 로직'이라는 개념이 아직도 명확하게 정립되지 않은 것 같아서 고민했던 것 같습니다
예외를 처리하는 방식 또한 서비스의 비즈니스 정책과 관련이 있으면 서비스 계층에서 처리하는 것이 SRP원칙에 부합하다고 생각합니다.

Rok93 commented on src/main/java/gift/product/service/ProductService.java:
```diff
@@ -27,6 +27,7 @@ public ProductService(ProductRepository productRepository, ProductValidator prod
 
   public PagedResult<GetProductResponseDto> getAllByPage(PageRequest pageRequest)
       throws IllegalArgumentException {
+    productValidator.validateProductSortField(pageRequest.sortInfo().field());
```
1단계 관련 코멘트로도 생각보다 할 이야기가 많고 또 2단계 미션이 `페이지네이션` 이기 때문에 이 부분은 2단계 미션 진행시에 관련 코멘트를 남기는 것으로 할께요. 😉 
원준님도 이 부분은 2단계 진행 시 저한테 다시 언급 부탁드려요!

가능하면 해당 코멘트 링크를 같이 언급해주시면 제가 히스토리 파악이 더 빠를 것 같아요. 🙏

lwj0831 replied:
넵 다음단계에서 피드백 부탁드립니다!

Rok93 commented on src/main/java/gift/product/repository/JpaProductRepositoryAdapter.java:
```diff
@@ -0,0 +1,60 @@
+  @Override
+  public void update(Long id, Product updateProduct) {
+    ProductEntity productEntity = jpaProductRepository.findById(id).orElseThrow(
+        ProductNotFoundException::new);
```
조회했을 때, 없는 product의  id가 무엇인지 예외 메시지로 보여준다면, 디버깅하기가 더 수월하지 않을까요? 😃

lwj0831 replied:
좀 더 명확한 예외 정보를 알 수 있을 것 같아 도움이 될 것 같습니다! 해당 내용 반영해서 리팩토링했습니다

Rok93 commented on src/main/java/gift/product/repository/JpaProductRepositoryAdapter.java:
```diff
@@ -0,0 +1,60 @@
+
+  @Override
+  public List<Product> findAllByPage(int offset, int pageSize, SortInfo sortInfo) {
+    Direction direction = sortInfo.isAscending() ? Direction.ASC : Direction.DESC;
```
현재 `정렬`에 대한 옵션으로는 (오름차순인지 여부로) true, false를 받고있는데요. 
애초에 `"asc"`, `"desc"` 와 같이 문자열을 받는다면 아래와 같이 변경할 수 있지 않을까요? 

```suggestion
    Direction direction =  Direction.fromString(sortInfo.sort());
```

혹은 애초에 "ASC", "DESC"로 정렬 요청 값을 받는다면,  아래와 같이 애초에 요청 파라미터를 `Direction` 타입으로 받을 수도 있을 것 같아요. 

```suggestion
     Direction direction = sortInfo.sort();
```

lwj0831 replied:
Spring data jpa를 사용하지 않고 있을 때 페이지네이션 기능을 구현했다가 점차 리팩토링하면서 애매모호해진 것 같습니다😂
좀 더 명확하게 처음부터 MethodArgumenteResolver에서 Direction을 검증 및 기본값 처리 후 받아도록 리팩토링했습니다!

Rok93 commented on src/main/java/gift/wishlist/repository/JpaWishItemRepository.java:
```diff
@@ -0,0 +1,26 @@
+  List<WishItemEntity> findAllByMemberId(@Param("memberId") Long memberId);
+
+  @Query("SELECT new gift.wishlist.dto.SimpleWishItemDto(pe.id, pe.name, pe.price, pe.imageUrl) FROM WishItemEntity we INNER JOIN ProductEntity pe ON we.product.id = pe.id WHERE we.member.id = :memberId")
+  List<SimpleWishItemDto> findWishItemsWithProductByMemberId(@Param("memberId") Long memberId);
```
DTO는 화면에 Fit한 정보라고 볼 수 있는데, Repository에서부터 일일이 DTO 에 맞춰서 조회를해주게되면 Repository가 점점 UI와 밀접해지지 않을까요? 

추가로 SimpleWishItemDto가 가진 정보에 추가적으로 member의 name도 필요한 API가 추가된다면  비스무리한 메서드가 또 하나 추가되지 않을까요? 🤔

lwj0831 replied:
레포지토리가 DTO에 의존되는 상황이라 현재 코드가 확신이 안서긴 했습니다!  제가 알기론 DTO를 반환하는 방식은 필요한 데이터만 바로 가져오므로 성능상의 이점 정도가 있는 걸로 알고 있습니다. 하지만 제가 느끼기에도 API 스펙에 맞춘 레포지토리가 되어서 재사용성이 낮아 리팩토링해야겠다고 생각하고 있었습니다 🤔 관련해서 리팩토링하였습니다

Rok93 reviewed (changes requested):
원준님 1단계 구현 빠르게 잘해주셨네요. 👍
현재 default profile 기준으로 애플리케이션이 실행되지 않고 테스트 코드도 깨지고있는 것 같은데, 확인해서 수정 부탁드릴게요. 🙏
질문 주신 내용과 몇몇 코멘트 남겨두었는데요. 확인 부탁드릴게요. 🤗

lwj0831 committed:
style: Code reformatting 들여쓰기 4로 변경 및 전체 코드 적용

lwj0831 committed:
refactor: 프로파일별 애플리케이션 설정 재구성

- default, test 두가지로 간단하게 분리
- default(MYSQL), test(H2)로 구성

lwj0831 committed:
refactor(member): Member 도메인, 엔티티 클래스 통합

- Jdbc기반 레포지토리, 헬퍼 클래스 삭제
- 기존의 도메인, 엔티티 분리 구조에서 Member클래스를 도메인, 엔티티로 사용
- 변경 사항 반영하여 Service로직 수정

lwj0831 committed:
refactor(product): Product 도메인, 엔티티 클래스 통합

    - Jdbc기반 레포지토리, 헬퍼 클래스 삭제
    - 기존의 도메인, 엔티티 분리 구조에서 Product클래스를 도메인, 엔티티로 사용
    - 변경 사항 반영하여 Service로직 수정

lwj0831 committed:
refactor(wishlist): WishItem 도메인, 엔티티 클래스 통합

        - Jdbc기반 레포지토리, 헬퍼 클래스 삭제
        - 기존의 도메인, 엔티티 분리 구조에서 WishItem클래스를 도메인, 엔티티로 사용
        - 변경 사항 반영하여 Service로직 수정

lwj0831 committed:
refactor(auth): MemberAuth 도메인, 엔티티 클래스 통합

            - Jdbc기반 레포지토리, 헬퍼 클래스 삭제
            - 기존의 도메인, 엔티티 분리 구조에서 MemberAuth클래스를 도메인, 엔티티로 사용
            - 변경 사항 반영하여 Service로직 수정

lwj0831 committed:
chore(test): 도메인별 Jdbc레포지토리 테스트 삭제

lwj0831 committed:
refactor: Pagination 로직 수정

- SortInfo에서 정렬 오름차순 여부를 boolean이 아닌 Direction클래스를
  받도록 수정
- MethodArgumentResolve에서 검증 및 기본값 처리하도록 유지

lwj0831 committed:
feat: 도메인별 NotFoundException발생 시 id 및 식별값 정보 로그
출력하도록 개선

- BuisnessException - DomainNotFoundException - 도메인별 NotFoundException 의 구조로 구현
- DomainNotFoundException에서 관련 로직 공통으로 처리

lwj0831 committed:
refactor(wishlist): WishItemJpaRepository DTO의존 구조 변경

- WishItem엔티티를 반환 후 연관 Product엔티티 정보 조회하는 방식으로
  변경

lwj0831 committed:
test(wishlist): WishItemJpaRepository 단위 테스트 추가

lwj0831 committed:
chore: 불필요한 ActiveProfiles 제거

lwj0831 committed:
feat: MemberAuth에 등록시간, 수정시간 포함

- JPA Auditing 사용하여 엔티티 관련 이벤트 발생 시 이벤트 리스너를 통해
  엔티티 등록, 수정 시간 정보 저장

Rok93 reviewed (approved):
원준님 피드백 반영 잘해주셨네요. 👍👍 
사실 1단계 미션은 충분히 잘해주셨지만 엔티티와 도메인 객체를 분리하신 부분에 대한 이야기를하다보니 조금 리뷰 일정이 지연된 듯 하네요.  (그래도 유익했...죠? 🙄)

1단계는 이만 머지하겠습니다. 
다음 단계에서 뵐게요! 🤗

Rok93 commented:
아...! @lwj0831 님  현재 애플리케이션 동작이 안되고있으니, 요것도 확인해서 동작하도록 수정해주세요!

lwj0831 commented:
> 아...! @lwj0831 님 현재 애플리케이션 동작이 안되고있으니, 요것도 확인해서 동작하도록 수정해주세요!

@Rok93 리팩도링 이후 제 환경에서는 잘 동작하는 것 같은데 혹시 어떤 부분에서 동작이 안되고 있는지 알 수 있을까요?
default, test 프로파일 환경 모두 정상 동작 확인했는데 어디서 안되고 있는지 모르겠어서요..!
